validator-api/src/rewarding/mod.rs

```diff
@@ -54,6 +55,19 @@ pub(crate) struct MixnodeToReward {
 
     /// Total number of individual addresses that have delegated to this particular node
     pub(crate) total_delegations: usize,
+    /// Node absolute uptime over total active set uptime
+    params: Option<NodeRewardParams>,
+}
+
+impl MixnodeToReward {
+    /// Somewhat clumsy way of feature gatting tokenomics payments. In a tokenomics scenario this will never be None at reward time. We levarage that to Into a different ExecuteMsg variant
```

Comment:
I'd perhaps argue that you shouldn't need to worry about `tokenomics` feature and you should just completely remove old rewards and replace it with your scheme.

Reply:
Yeah, although I'd make it into a separate PR

Reply:
Created an issue for this